sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/csv/CSVTable.scala

```diff
@@ -37,8 +37,13 @@ case class CSVTable(
     userSpecifiedSchema: Option[StructType],
     fallbackFileFormat: Class[_ <: FileFormat])
   extends FileTable(sparkSession, options, paths, userSpecifiedSchema) {
-  override def newScanBuilder(options: CaseInsensitiveStringMap): CSVScanBuilder =
-    CSVScanBuilder(sparkSession, fileIndex, schema, dataSchema, options)
+  override def newScanBuilder(options: CaseInsensitiveStringMap): CSVScanBuilder = {
+    val finalOptions = this.options.asCaseSensitiveMap().asScala.filterNot {
+      case (k, _) => options.containsKey(k)
```

Comment:
do we need this? I think combining two maps will respect the keys in the later map.